src/vision/rust_image.rs

```diff
@@ -0,0 +1,180 @@
+    ///  `1 * 3 * h * w` => `h * w * 3`
+    fn try_from(value: &'i Tensor) -> Result<Self, Self::Error> {
+        let tensor = assert_tensor_as_image(value, Kind::Uint8, 3)?;
+        let width = tensor.size()[1] as u32;
+        let height = tensor.size()[0] as u32;
+        let length = (width * height * 3) as usize;
+        let mut buffer = vec![0; length];
+        tensor.f_copy_data(&mut buffer, length)?;
+        match RgbImage::from_raw(width, height, buffer) {
+            Some(s) => Ok(s),
+            None => Err(TchError::Convert("Failed to convert tensor to image".to_string())),
+        }
+    }
+}
+
+impl<'i> From<&'i Rgb32FImage> for Tensor {
+    /// `h * w * 3` => `1 * 3 * h * w`
+    fn from(rgb: &'i Rgb32FImage) -> Self {
+        let kind = Kind::Float;
+        let size = &[rgb.height() as i64, rgb.width() as i64, 3];
+        let tensor =
+            Tensor::f_of_data_size(rgb.as_bytes(), size, kind).expect("Failed to create tensor");
+        hwc_to_chw(&tensor)
+    }
+}
+
+impl<'i> From<&'i Rgba32FImage> for Tensor {
+    /// `h * w * 4` => `1 * 4 * h * w`
+    fn from(rgb: &'i Rgba32FImage) -> Self {
+        let kind = Kind::Float;
+        let size = &[rgb.height() as i64, rgb.width() as i64, 4];
+        let tensor =
+            Tensor::f_of_data_size(rgb.as_bytes(), size, kind).expect("Failed to create tensor");
+        hwc_to_chw(&tensor)
+    }
+}
+
+impl<'i> TryFrom<&'i Tensor> for Rgb32FImage {
+    type Error = TchError;
+    ///  `1 * 3 * h * w` => `h * w * 3`
+    fn try_from(value: &'i Tensor) -> Result<Self, Self::Error> {
+        let tensor = assert_tensor_as_image(value, Kind::Float, 3)?;
+        let width = tensor.size()[1] as u32;
+        let height = tensor.size()[0] as u32;
+        let length = (width * height * 3) as usize;
+        let mut buffer = vec![0.0; length];
+        tensor.f_copy_data(&mut buffer, length)?;
+        match Rgb32FImage::from_raw(width, height, buffer) {
+            Some(s) => Ok(s),
+            None => Err(TchError::Convert("Failed to convert tensor to image".to_string())),
+        }
+    }
+}
+
+#[inline]
+fn assert_tensor_as_image(tensor: &Tensor, except: Kind, channel: i64) -> Result<Tensor, TchError> {
+    let kind = tensor.kind();
+    let size = tensor.size();
+    if size.len() != 3 {
+        let msg = format!("Tensor size is `{size:?}`, except rank 3 tensor", size = size);
+        Err(TchError::Convert(msg))?;
+    }
+    if size[0] != channel {
+        let msg = format!(
+            "Tensor size is `{size:?}`, except {channel} channels",
+            size = size,
+            channel = channel
+        );
+        Err(TchError::Convert(msg))?;
+    }
+    if kind != except {
+        let msg = format!(
+            "Tensor kind is `{kind:?}`, except `{except:?}` tensor",
+            kind = kind,
+            except = except
+        );
+        Err(TchError::Convert(msg))?;
+    }
+    Ok(chw_to_hwc(tensor))
+}
+
+// #[test]
```

Comment:
Could you either enable or remove these tests?

Reply:
There is no suitable image in `tests/`, if there is, I can move to the `tests/` directory.